File: programs/hydra-staking/src/state/pool_state.rs
use anchor_lang::prelude::*;

#[account]
#[derive(Debug, Default)]
pub struct PoolState {
    pub authority: Pubkey,
    pub token_vault: Pubkey,
    pub token_mint: Pubkey,
    pub token_mint_decimals: u8,
    pub redeemable_mint: Pubkey,
    pub pool_state_bump: u8,
    pub token_vault_bump: u8,
}
impl PoolState {}

File: sdks/hydra-ts/src/staking/types/pool-state.ts
import { PublicKey } from "@solana/web3.js";

export type PoolState = {
  authority: PublicKey;
  tokenVault: PublicKey;
  tokenMint: PublicKey;
  tokenMintDecimals: number;
  redeemableMint: PublicKey;
  poolStateBump: number;
  tokenVaultBump: number;
};
